### PR TITLE
DROOLS-6987 Sonarqube based fixes

### PR DESCRIPTION
**Pull request by gorky**
**JIRA**:
[https://issues.redhat.com/browse/DROOLS-6987](https://issues.redhat.com/browse/DROOLS-6987)

https://jenkins.updater.j2eeguys.com/job/Kiegroup%20Drools

NOTE: Commit log was too large to include.  See [https://github.com/kiegroup/drools/commit/fc0892a0d8173eaba0cbcadfd83f9af0425a6446](https://github.com/kiegroup/drools/commit/fc0892a0d8173eaba0cbcadfd83f9af0425a6446)

Jenkins execute product fdb

### CODE REVIEW

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2010 salaboy.
+ * Copyright 2022 salaboy.
```
Comment (tarilabs) [inline drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java]: As an interesting side-effect, this was brought to daylight :) 
To be explicit I don't think any malice was intended, just a wrong configuration of the IDE and Tooling used at the time to insert header.

File: drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java [hunk 2]
```diff
@@ -83,8 +83,8 @@ private Object recursiveMarshal( Object o, Map<Object, Object> seenObjectsMap )
                 return new JaxbListWrapper(serializedArr, JaxbWrapperType.SET);
             } else if( o instanceof Map ) {
                 Map<Object, Object> map = (Map<Object, Object>) o;
-                List<JaxbStringObjectPair> pairList = new ArrayList<JaxbStringObjectPair>(map.size());
-                if( map == null || map.isEmpty() ) {
```
Comment (tarilabs) [inline drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java:87]: this had no mention in the JIRA, but looks okay to me (ref instanceof check)

**Review by mariofusco — CHANGES_REQUESTED** [review]
We never change the year in the license header. Please revert those changes.

**Review by tarilabs — COMMENTED** [review]
Thank you for the PR !
From my side, I've looked and I couldn't find changes relating to `File.toURL()` or derivatives, which are mentioned in the JIRA, but indeed coudln't find them here?
Some other things I've noticed below.

Others might have additional comments.

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java (hunk 1 above)
Comment (gorky) [reply drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java]: Yeah, the next release will APPEND the current year to the last copyright entry in the file per the FSF, not update the existing year.  For the FSF, they state "You should add the proper year for each past release" . See [https://www.gnu.org/licenses/gpl-howto.html](https://www.gnu.org/licenses/gpl-howto.html) for details.

I made some assumptions
1. that there was only one set of copyrights in a file (obviously that's wrong and I am fixing that on a priority basis)
2. the current maintainers would have their copyrights in the file
3. that all projects need the copyrights updated when a file was modified.

My apologies for any errors, and let me know if I need to go fix anything.

-Steve

File: drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java (hunk 2 above)
Comment (gorky) [reply drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java:87]: Unfortunately GitHub seems to be trimming the comment to the first 64K, and I missed that when I was scanning the log.   My apologies for the confusion.  For reference, Sonar has it as 'squid:S2589', [https://rules.sonarsource.com/java/RSPEC-2589](https://rules.sonarsource.com/java/RSPEC-2589)

**Comment by gorky** [comment]
@mariofusco -- Will do on the changes for the Headers.  Might take me a day or two to setup.

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java (hunk 1 above)
Comment (gorky) [reply drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java]: And I just realized that I misread your comment -- I thought it was my tool's modification that was the issue... Whoops!  As for what happened with the Updater, I am hoping that you might find this info useful....

-Steve

Comment (tarilabs) [reply drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java]: This PR is useful to address Sonar highlights Steve, thank you.
Indeed, I was only making note that I noticed this, which was previously existing.

As Mario mentioned however, we don't need header modifications.

File: drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java (hunk 2 above)
Comment (gorky) [reply drools-commands/src/main/java/org/drools/commands/jaxb/JaxbUnknownAdapter.java:87]: Forgot to mention, the full update log with what files were modified and why is in the Git commit and visible in the Git Log.

**Comment by gorky** [comment]
Updated changes submitted w/o Copyright/Header changes.

**Review by mariofusco — APPROVED** [review]

**Comment by gorky** [comment]
Are the Kogito issues from changes submitted with this PR?  Or external to this activity?  Specifically, is this something I caused and need to fix, or a different issue?

**Comment by mariofusco** [comment]
@gorky nope, not related with this PR, just some flaky test I guess, trying to figure out what's going on.

**Pull request merged**